op-node/node/node.go

```diff
@@ -553,6 +551,20 @@ func (n *OpNode) OnUnsafeL2Payload(ctx context.Context, from peer.ID, envelope *
 		return nil
 	}
 
+	// During sequencer HA setup, there is an edge case that could happen:
+	// 1. Sequencer A is the cluster leader and sequencing
+	// 2. op-conductor on Sequencer A went down
+	// 3. Sequencer B took over and started sequencing via leader election
+	// 4. At this moment, Sequencer A is still running and in sequencing mode (op-conductor down, cannot turn off sequencing automatically on A)
+	// 5. Sequencer A received a unsafe payload from B
+	// In this case, we should ignore the payload from B, because current sequencer is stuck at RunNextSequencerAction step
+	// trying to build on top of the last unsafe block it produced but keeps failing (due to cannot commit unsafe payload to consensus layer).
+	// However if we allow the payload from B to be processed, it will cache the payload on op-node and also create a block on op-geth.
+	// It would be better to ignore the payload from B entirely and keep the sequencer in stuck state until it's remediated.
+	if active, _ := n.l2Driver.SequencerActive(ctx); active {
```

Comment:
The alternative is to allow this to go through and process the blocks. It won't cause network reorg, but would make the sequencer state seems a little weird (cannot produce blocks, but blocks keep increasing)

curious your thoughts on this? @protolambda @mslipper @trianglesphere

Reply:
This PR is mainly for discussion purposes, if we agree this is necessary, I'll include it in https://github.com/ethereum-optimism/optimism/pull/8940/files, and there's e2e test cases to cover this